js2-mode.el

```diff
@@ -8166,20 +8301,29 @@ arrow function), NAME is js2-name-node."
         ;;    of the with object.
         (setf (js2-function-node-ignore-dynamic fn-node) t))
     ;; dynamically bind all the per-function variables
-    (let ((js2-current-script-or-fn fn-node)
-          (js2-current-scope fn-node)
-          (js2-nesting-of-with 0)
-          (js2-end-flags 0)
-          js2-label-set
-          js2-loop-set
-          js2-loop-and-switch-set)
-      (js2-parse-function-params function-type fn-node pos)
+    (let* ((js2-current-script-or-fn fn-node)
+           (js2-current-scope fn-node)
+           (js2-nesting-of-with 0)
+           (js2-end-flags 0)
+           js2-label-set
+           js2-loop-set
+           js2-loop-and-switch-set
+           (error-checking-arguments (js2-parse-function-params
+                                      function-type fn-node pos)))
       (when (eq function-type 'FUNCTION_ARROW)
         (js2-must-match js2-ARROW "msg.bad.arrow.args"))
       (if (and (>= js2-language-version 180)
                (/= (js2-peek-token) js2-LC))
           (js2-parse-function-closure-body fn-node)
         (js2-parse-function-body fn-node))
+      (when (js2-function-node-in-strict-mode fn-node)
```

Comment:
Couldn't you perform this logic while `js2-in-use-strict-directive` is still assigned? Then there will be no need for the `in-strict-mode` slot.